providers/src/airflow/providers/edge/executors/edge_executor.py

```diff
@@ -86,9 +88,30 @@ def execute_async(
             )
         )
 
-    @provide_session
-    def sync(self, session: Session = NEW_SESSION) -> None:
-        """Sync will get called periodically by the heartbeat method."""
+    def _check_worker_liveness(self, session: Session) -> bool:
+        """Reset worker state if heartbeat timed out."""
+        changed = False
+        heartbeat_interval: int = conf.getint("edge", "heartbeat_interval")
+        lifeless_workers: list[EdgeWorkerModel] = (
+            session.query(EdgeWorkerModel)
+            .filter(
+                EdgeWorkerModel.state != EdgeWorkerState.UNKNOWN
+                and EdgeWorkerModel.state != EdgeWorkerState.OFFLINE
+                and EdgeWorkerModel.last_update
+                < (timezone.utcnow() - timedelta(seconds=heartbeat_interval * 5))
```

Comment:
I believe the ORM mapper rather wants it as array of filters like the following?
```suggestion
                EdgeWorkerModel.state.not_in(EdgeWorkerState.UNKNOWN, EdgeWorkerState.OFFLINE),
                EdgeWorkerModel.last_update < (timezone.utcnow() - timedelta(seconds=heartbeat_interval * 5))
```

Reply:
EdgeWorkerModel.state.not_in is not working. I changed the code to a different check.

Reply:
Fixed it now